python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -365,9 +365,17 @@ def _create_node(self, node_config, tags, count):
                 conf.update({
                     "MinCount": 1,
                     "MaxCount": count,
-                    "SubnetId": subnet_id,
+                    "NetworkInterfaces": [{
+                        "AssociatePublicIpAddress": True,
+                        "DeviceIndex": 0,
+                        "SubnetId": subnet_id,
+                    }],
                     "TagSpecifications": tag_specs
                 })
+                maybe_security_group_ids = conf.pop("SecurityGroupIds", None)
```

Comment:
nit: I find just `security_group_ids` clearer (especially since it's compared to None in the next line)

Reply:
(matter of taste, of course)

Reply:
You have fine taste